### PR TITLE
NMS-16284: User Data Collection CM, liquibase, Rest updates

### PR DESCRIPTION
**Pull request by synqotik**
User Data Collection updates to Configuration Management including the liquibase schema update. Added REST endpoint for getting and changing status of whether someone acknowledged the UI notice and opted in or out.

Note, we are only keeping track of whichever user acknowledged the notice first.

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16284

### CODE REVIEW

File: features/config/upgrade/src/main/resources/changelog-cm/33.0.0/changelog-cm.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:cm="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd
+    http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd">
+
+    <changeSet author="stheleman" id="33.0.0-update-schema-datachoices-with-user-data-collection">
+        <cm:changeSchema schemaId="org.opennms.features.datachoices">
+            <cm:put name="userDataCollectionOptedIn" type="boolean" default="false"/>
+            <cm:put name="userDataCollectionNoticeAcknowledged" type="boolean" default="false"/>
+            <cm:put name="userDataCollectionNoticeAcknowledgedAt" type="string"/>
```
Comment (RangerRick) [inline features/config/upgrade/src/main/resources/changelog-cm/33.0.0/changelog-cm.xml:13]: do we not have a date type in the CM stuff?

Comment (synqotik) [reply features/config/upgrade/src/main/resources/changelog-cm/33.0.0/changelog-cm.xml:13]: From what I can tell in the following, only boolean, string and number were implemented:

https://github.com/OpenNMS/opennms/tree/develop/features/config/upgrade/src/main/java/liquibase/ext2/cm/change/types

and `propertyType` in
https://github.com/OpenNMS/opennms/blob/develop/features/config/upgrade/src/main/resources/liquibase.parser.core.xml/dbchangelog-ext.xsd

File: features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java [hunk 2]
```diff
@@ -99,6 +103,39 @@ public Response setStatus(HttpServletRequest request, UsageStatisticsStatusDTO d
...
+
+            if (dto.getOptedIn() != null) {
+                m_stateManager.setUserDataCollectionOptedIn(dto.getOptedIn());
+            }
+
+            if (dto.getNoticeAcknowledged() != null) {
+                m_stateManager.setUserDataCollectionNoticeAcknowledged(dto.getNoticeAcknowledged(), remoteUser);
+            }
+        } catch (Exception e) {
+            throw Throwables.propagate(e);
```
[the remark this answers is not in the document]
Comment (synqotik) [reply features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java:133]: Good point, might make sense to change all of these in this file / REST service

[the remark this answers is not in the document]
Comment (synqotik) [reply features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java]: Updated to match what most REST v2 APIs are doing.

File: features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java [hunk 3]
```diff
@@ -120,4 +162,9 @@ public void setStateManager(StateManager stateManager) {
     public void setUsageStatisticsReporter(UsageStatisticsReporter usageStatisticsReporter) {
         m_usageStatisticsReporter = Objects.requireNonNull(usageStatisticsReporter);
     }
+
+    private WebApplicationException getException(String msg, Throwable e) {
```
Comment (cgorantla) [inline features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java:166]: I think you can omit `WebApplicationException` and directly return Response.  Is there any advantage of using `WebApplicationException` ?

Comment (synqotik) [reply features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java:166]: Not really sure ... just following pattern in many other v2 REST services. I'll change to just return Response and do the logging.

Comment (synqotik) [reply features/datachoices/src/main/java/org/opennms/features/datachoices/web/internal/DataChoiceRestServiceImpl.java]: Updated to log exception and return `Response`.

**Review by RangerRick — APPROVED** [review]

**Pull request merged**